k8s/namespace_deleter.go

```diff
@@ -0,0 +1,18 @@
+package k8s
+
+// NamespaceDeleter is a (k8s.io/kubernetes/pkg/client/unversioned).NamespaceInterface compatible
+// interface designed only for deleting namespaces. It should be used as a parameter to functions
+// so that they can be more easily unit tested
+type NamespaceDeleter interface {
+	Delete(name string) error
+}
+
+// FakeNamespaceDeleter is a NamespaceDeleter implementation to be used in unit tests
+type FakeNamespaceDeleter struct {
+	Err error
```

Comment:
shouldn't `FakeNamespaceDeleter` keep a list of the namespaces that have been deleted? tests can then check that list to ensure that the right namespaces have been deleted